tests/integration/security/ca_custom_root/trust_domain_alias_secure_naming_test.go

```diff
@@ -0,0 +1,242 @@
+						},
+					},
+					ServiceAccount: true,
+					Ports: []echo.Port{
+						{
+							Name:         HTTPS,
+							Protocol:     protocol.HTTPS,
+							ServicePort:  443,
+							InstancePort: 8443,
+							TLS:          true,
+						},
+					},
+					TLSSettings: &common.TLSSettings{
+						RootCert:   loadCert(t, "root-cert.pem"),
+						ClientCert: loadCert(t, "tmp/workload-foo-cert.pem"),
+						Key:        loadCert(t, "tmp/workload-foo-key.pem"),
+					},
+				}).
+				With(&serverNakedBar, echo.Config{
+					Namespace: testNS,
+					Service:   "server-naked-bar",
+					Subsets: []echo.SubsetConfig{
+						{
+							Annotations: echo.NewAnnotations().SetBool(echo.SidecarInject, false),
+						},
+					},
+					ServiceAccount: true,
+					Ports: []echo.Port{
+						{
+							Name:         HTTPS,
+							Protocol:     protocol.HTTPS,
+							ServicePort:  443,
+							InstancePort: 8443,
+							TLS:          true,
+						},
+					},
+					TLSSettings: &common.TLSSettings{
+						RootCert:   loadCert(t, "root-cert.pem"),
+						ClientCert: loadCert(t, "tmp/workload-bar-cert.pem"),
+						Key:        loadCert(t, "tmp/workload-bar-key.pem"),
+					},
+				}).
+				BuildOrFail(t)
+
+			ctx.Config().ApplyYAMLOrFail(ctx, testNS.Name(), fmt.Sprintf(POLICY))
+
+			verify := func(t *testing.T, src echo.Instance, dest echo.Instance, s scheme.Instance, success bool) {
+				t.Helper()
+				want := "success"
+				if !success {
+					want = "fail"
+				}
+				name := fmt.Sprintf("server:%s[%s]", dest.Config().Service, want)
+				t.Run(name, func(t *testing.T) {
+					t.Helper()
+					opt := echo.CallOptions{
+						Target:   dest,
+						PortName: HTTPS,
+						Address:  dest.Config().Service,
+						Scheme:   s,
+					}
+					checker := connection.Checker{
+						From:          src,
+						Options:       opt,
+						ExpectSuccess: success,
+					}
+					checker.CheckOrFail(ctx)
+				})
+			}
+
+			cases := []struct {
+				src    echo.Instance
+				dest   echo.Instance
+				expect bool
+			}{
+				{
+					src:    client,
+					dest:   serverNakedFoo,
+					expect: true,
+				},
+				{
+					src:    client,
+					dest:   serverNakedBar,
+					expect: false,
+				},
+			}
+
+			for _, tc := range cases {
+				verify(t, tc.src, tc.dest, scheme.HTTP, tc.expect)
+			}
+		})
+}
+
+func loadCert(t test.Failer, name string) string {
+	data, err := cert.ReadSampleCertFromFile(name)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return string(data)
+}
+
+func generateCerts(t test.Failer, ns string) func() {
+	workDir := path.Join(env.IstioSrc, "samples/certs")
+	script := path.Join(workDir, "generate-workload.sh")
+
+	crts := []struct {
+		td string
+		sa string
+	}{
+		{
+			td: "foo",
+			sa: "server-naked-foo",
+		},
+		{
+			td: "bar",
+			sa: "server-naked-bar",
+		},
+	}
+
+	for _, crt := range crts {
+		command := exec.Cmd{
+			Path:   script,
+			Args:   []string{script, crt.td, ns, crt.sa, workDir, "tmp"},
```

Comment:
use a constant var to refer to it? This hardcoded value is used multiple times in the code?